blockdag/process.go

```diff
@@ -168,6 +168,24 @@ func (dag *BlockDAG) ProcessBlock(block *util.Block, flags BehaviorFlags) (isOrp
 		}
 	}
 
+	// Handle the case of a block with a valid timestamp(non-delayed) which points to a delayed block.
+	dag.delayedBlocksLock.RLock()
+	delay = 0
+	for _, parentHash := range block.MsgBlock().Header.ParentHashes {
```

Comment:
This check should only happen after base check that all parents exist, and loop only over parents that don't exist